pkg/cluster/docker_for_mac.go

```diff
@@ -0,0 +1,199 @@
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// Uses the DockerForMac GUI protocol to control DockerForMac.
+type DockerForMacClient struct {
+	httpClient HTTPClient
+	socketPath string
+}
+
+func NewDockerForMacClient() (DockerForMacClient, error) {
+	homedir, err := homedir.Dir()
+	if err != nil {
+		return DockerForMacClient{}, err
+	}
+
+	socketPath := filepath.Join(homedir, "Library/Containers/com.docker.docker/Data/gui-api.sock")
+	httpClient := &http.Client{
+		Transport: &http.Transport{
+			DialContext: func(ctx context.Context, _, _ string) (net.Conn, error) {
+				return net.Dial("unix", socketPath)
+			},
+		},
+	}
+	return DockerForMacClient{
+		httpClient: httpClient,
+		socketPath: socketPath,
+	}, nil
+}
+
+func (c DockerForMacClient) writeSettings(ctx context.Context, settings map[string]interface{}) error {
+	klog.V(7).Infof("POST %s /settings\n", c.socketPath)
+	buf := bytes.NewBuffer(nil)
+	err := json.NewEncoder(buf).Encode(c.settingsForWrite(settings))
+	if err != nil {
+		return errors.Wrap(err, "writing d4m settings")
+	}
+
+	klog.V(8).Infof("Request body: %s\n", buf.String())
+	req, err := http.NewRequest("POST", "http://localhost/settings", buf)
+	if err != nil {
+		return errors.Wrap(err, "writing d4m settings")
+	}
+
+	req.Header.Add("Content-Type", "application/json")
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return errors.Wrap(err, "writing d4m settings")
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusCreated {
+		return fmt.Errorf("writing d4m settings: status code %d", resp.StatusCode)
+	}
+	return nil
+}
+
+func (c DockerForMacClient) settings(ctx context.Context) (map[string]interface{}, error) {
+	klog.V(7).Infof("GET %s /settings\n", c.socketPath)
+	req, err := http.NewRequest("GET", "http://localhost/settings", nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "reading d4m settings")
+	}
+
+	resp, err := c.httpClient.Do(req)
+	if err != nil {
+		return nil, errors.Wrap(err, "reading d4m settings")
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("reading d4m settings: status code %d", resp.StatusCode)
+	}
+
+	settings := make(map[string]interface{})
+	err = json.NewDecoder(resp.Body).Decode(&settings)
+	if err != nil {
+		return nil, errors.Wrap(err, "reading d4m settings")
+	}
+	klog.V(8).Infof("Response body: %+v\n", settings)
+	return settings, nil
+}
+
+func (c DockerForMacClient) ensureK8sEnabled(settings map[string]interface{}) (changed bool, err error) {
+	vmSetting, ok := settings["vm"].(map[string]interface{})
+	if !ok {
+		return false, fmt.Errorf("missing vm setting")
+	}
+
+	k8sSetting, ok := vmSetting["kubernetes"].(map[string]interface{})
+	if !ok {
+		return false, fmt.Errorf("missing kubernetes setting")
+	}
+
+	enabledSetting, ok := k8sSetting["enabled"].(map[string]interface{})
+	if !ok {
+		return false, fmt.Errorf("missing kubernetes enabled setting")
```

Comment:
for extra clarity, might give the full path her: "missing vm.kubernetes.enabled setting" etc.

Reply:
done!